src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/ManagedStrategy.kt

```diff
@@ -731,6 +734,7 @@ abstract class ManagedStrategy(
             lastReadTracePoint[iThread] = tracePoint
         }
         newSwitchPoint(iThread, codeLocation, tracePoint)
+        loopDetector.passValue(obj)
```

Comment:
The fact that loop detector uses specific values to detect spin-loops is an algorithm's implementation detail. I think we can design the `LoopDetector` API to be a bit cleaner (and more flexible for potential future changes). 

Let's instead design the API in way, so that `LoopDetector` just listens for the specific events in the execution. For example, you already have the following methods in the `LoopDetector` API:
```
visitCodeLocation
beforeRegularMethodCall
afterRegularMethodCall
```

So let's just extend this API similarly for other types of events:
```
beforeReadField
beforeReadArrayElement
afterRead
...
```

Then in the implementation of these methods in `LoopDetector` you hash only the arguments that are required by the algorithm.

--

P.S.
Ideally, we should eventually converge to a design when different components of the managed strategy just implement `EventTracker` interface and subscribe for the events they are interested in (this is just an idea for the future, not for this PR).